datafusion/core/src/datasource/physical_plan/parquet/mod.rs

```diff
@@ -805,6 +806,16 @@ impl ExecutionPlan for ParquetExec {
             .clone()
             .unwrap_or_else(|| Arc::new(DefaultSchemaAdapterFactory));
 
+        let file_decryption_properties: Option<Arc<FileDecryptionProperties>> = match
+            &self.table_parquet_options.global.file_decryption_properties {
+            Some(d) => {
+                let fd =
+                    FileDecryptionProperties::builder(d.footer_key.as_bytes().to_vec()).build().unwrap();
+                Some(Arc::new(fd))}
+            ,
+            None => None
+        };
+
```

Comment:
This is the most important part where we pass the session options for decryption down to the parquet file opener.